src/OpenTelemetry.Api/Trace/TelemetrySpanNew.cs

```diff
@@ -0,0 +1,187 @@
+﻿// <copyright file="TelemetrySpanNew.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// <para>Span represents the execution of the certain span of code or span of time between two events which is part of
+    /// a distributed trace and has result of execution, context of execution and other properties.</para>
+    /// </summary>
+    public class TelemetrySpanNew : IDisposable
```

Comment:
@cijothomas If you do another round of changes would you mind tweaking some of the returns so we have a fluent(?) signature? See: https://github.com/open-telemetry/opentelemetry-dotnet/issues/705#issuecomment-637056216